src/Topshelf/Configuration/HostConfigurators/HostConfigurator.cs

```diff
@@ -116,5 +116,12 @@ public interface HostConfigurator
         /// <param name="name"></param>
         /// <param name="callback"></param>
         void AddCommandLineDefinition(string name, Action<string> callback);
+
+        /// <summary>
+        /// Adds custom exception handler that will be called for any exception caught
+        /// by Topshelf while a service is starting, running or stopping.
+        /// </summary>
+        /// <param name="callback"></param>
+        void AddExceptionHandler(Action<Exception> callback);

```

Comment:
Are you thinking we should have multiple handlers based on which operation (run, stop, unhandled, etc) the exception came from? Or perhaps calling it something like `OnException(...)`?